src/wallet/rpcwallet.cpp

```diff
@@ -2540,15 +2554,113 @@ static UniValue loadwallet(const JSONRPCRequest& request)
+    }
+
+    if (request.fHelp || request.params.size() > 1) {
+        std::string flags = "";
+        for (auto& it : walletflagmap) flags += (flags == "" ? "" : ", ") + it.first;
+        throw std::runtime_error(
+            "getwalletflags ( flag )\n"
+            "\nReturn the state of the (given) wallet flag(s) for a wallet.\n"
+            "\nArguments:\n"
+            "1. flag    (string, optional) The name of the flag. Current available flags: " + flags + "\n"
+            "\nResult (default):\n"
+            "{\n"
+            "  \"flagname\": true|false,  (string:bool pair) the name of the flag and its state"
+            "}\n"
+            "\nResult (flag=XXX)\n"
+            "true|false                 (bool) the state of the given flag\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getwalletflags", "")
+            + HelpExampleRpc("getwalletflag", "")
+        );
+    }
+
+    if (request.params[0].isNull()) {
+        UniValue res(UniValue::VOBJ);
+        for (const auto& it : walletflagmap) {
+            res.pushKV(it.first, pwallet->IsWalletFlagSet(it.second));
+        }
+
+        return res;
+    }
+
+    std::string flag = request.params[0].get_str();
+
+    if (!walletflagmap.count(flag)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Unknown wallet flag %s", flag));
+    }
+
+    return pwallet->IsWalletFlagSet(walletflagmap.at(flag));
+}
+
+static UniValue setwalletflag(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
     if (request.fHelp || request.params.size() < 1 || request.params.size() > 2) {
+        std::string flags = "";
+        for (auto& it : walletflagmap) flags += (flags == "" ? "" : ", ") + it.first;
+        throw std::runtime_error(
+            "setwalletflag flag ( state )\n"
+            "\nChange the state of the given wallet flag for a wallet.\n"
+            "\nArguments:\n"
+            "1. flag    (string, required) The name of the flag to change. Current available flags: " + flags + "\n"
+            "2. value   (boolean, optional, default=true) The new state.\n"
+            "\nResult:\n"
+            "true|false (boolean) True or false, reflecting whether the flag was modified (true), or (false) whether it was already set to the given state\n"
+            "\nExamples:\n"
+            + HelpExampleCli("setwalletflag", "avoidreuse")
+            + HelpExampleRpc("setwalletflag", "\"avoidreuse\"")
+        );
+    }
+
+    std::string flag = request.params[0].get_str();
+    bool value = request.params[1].isNull() || request.params[1].get_bool();
+
+    if (!walletflagmap.count(flag)) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Unknown wallet flag %s", flag));
+    }
+
+    if (walletflagmap.at(flag) >= (1ULL << 32)) {
```

Comment:
I don't think this is always necessarily going to be true. The meaning of the upper section/lower section flags are which we can safely ignore if we don't know the meaning of (ie if we open a newer wallet on an older client), not whether the flags are mutable or not.

Reply:
Ahh. Got it.